### PR TITLE
Allow more valid accept headers

### PR DESCRIPTION
**Pull request by ashokdelphia**
The main aim here is to allow more valid Accept headers, particularly types with other parameters than `q`.

I also was more permissive with optional whitespace arounnd the semi-colon, and allowed up to three decimal places but only in the inclusive range from zero to one.

I haven't added any negative tests (e.g. for `q=1.500` or with too many decimal digits), just for reasons of time, but can if they'd be desirable.

See #220.

### CODE REVIEW

**Comment by codingjoe** [comment]
@ashokdelphia this looks good. please also add the negative examples to make sure we cover all cases we can think of. Thanks again, I really appreciate the help!

File: health_check/views.py
Comment (ashokdelphia) [inline health_check/views.py:44]: I'm tempted to replace lines 38–44 with just:

```
          (\s*;|$)                        # followed by other parameters or the end of string                                                   
```

And not care about the structure of the other parameters. This would cause the `*/*;0.9` test to fail, but I wonder if that's preferable.

**Comment by ashokdelphia** [comment]
> @ashokdelphia this looks good. please also add the negative examples to make sure we cover all cases we can think of. 

Only in trying to add cases for invalid q-values, I noticed that they're still permitted by the last clause in the regular expression. I'm not sure it's worth making that more fierce in order to reject them.

I did add a few other failing cases, though, and left a structure where dropping more in is reasonably straightforward. 

> Thanks again, I really appreciate the help!

You're very welcome. Cheers.

File: health_check/views.py
Comment (codingjoe) [reply health_check/views.py:44]: Your call, but I can live with complex regular expressions.

**Review by codingjoe — APPROVED** [review]
LGTM, thanks @ashokdelphia

**Comment by codingjoe** [comment]
@ashokdelphia let me know how you decide on the open question you had. I will merge and release once you decided.

**Comment by ashokdelphia** [comment]
> @ashokdelphia let me know how you decide on the open question you had. I will merge and release once you decided.

Let's go with what we have, then. If we find we hit any other error cases, we can make it looser about the other parameters later.

**Comment by codingjoe** [comment]
Released in [3.10.1](https://github.com/KristianOellegaard/django-health-check/releases/tag/3.10.1)

**Pull request merged**
